xbmc/filesystem/posix/PosixFile.cpp

```diff
@@ -313,6 +314,18 @@ bool CPosixFile::Rename(const CURL& url, const CURL& urlnew)
   
   if (rename(name.c_str(), newName.c_str()) == 0)
     return true;
+
+  // rename across mount points - need to copy/delete
+  if (errno == EXDEV)
+  {
+    if (XFILE::CFile::Copy(name, newName))
+    {
+      if (XFILE::CFile::Delete(name))
```

Comment:
IMO file move should be treated as atomic operation, just like on a file system. If either step fails, the whole operation must fail, without leaving artifacts. We should not "silently" fall back to copy.